hardsigmoid operator for QNNPACK, using LUTs

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #36699 hardsigmoid: add PyTorch wrapper for the QNNPACK path
* **#36698 hardsigmoid operator for QNNPACK, using LUTs**

Summary:

Adds the hardsigmoid op to QNNPACK using the LUT kernel

Test Plan:

```
cd aten/src/ATen/native/quantized/cpu/qnnpack
with-proxy ./scripts/build-local.sh
./build/local/hardsigmoid-test
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21057153](https://our.internmc.facebook.com/intern/diff/D21057153)

vkuzo committed:
hardsigmoid operator for QNNPACK, using LUTs

Summary:

Adds the hardsigmoid op to QNNPACK using the LUT kernel

Test Plan:

```
cd aten/src/ATen/native/quantized/cpu/qnnpack
with-proxy ./scripts/build-local.sh
./build/local/hardsigmoid-test
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

z-a-f commented on aten/src/ATen/native/quantized/cpu/qnnpack/src/hardsigmoid.c:
```diff
@@ -0,0 +1,164 @@
+  if (!pytorch_qnnp_params.initialized) {
+    pytorch_qnnp_log_error(
+        "pytorch_qnnp_create_hardsigmoid_nc_q8 failed because QNNPACK is not properly initialized");
+    goto error;
```
`goto` is generally considered harmful: http://david.tribble.com/text/goto.html Consider replacing with 

```
pytorch_qnnp_delete_operator(hardsigmoid_op);
return pytorch_qnnp_status_uninitialized;
```

z-a-f reviewed (approved):
LGTM

vkuzo committed:
Update on "hardsigmoid operator for QNNPACK, using LUTs"


Summary:

Adds the hardsigmoid op to QNNPACK using the LUT kernel

Test Plan:

```
cd aten/src/ATen/native/quantized/cpu/qnnpack
with-proxy ./scripts/build-local.sh
./build/local/hardsigmoid-test
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21057153](https://our.internmc.facebook.com/intern/diff/D21057153)

[ghstack-poisoned]

vkuzo committed:
Update on "hardsigmoid operator for QNNPACK, using LUTs"


Summary:

Adds the hardsigmoid op to QNNPACK using the LUT kernel

Test Plan:

```
cd aten/src/ATen/native/quantized/cpu/qnnpack
with-proxy ./scripts/build-local.sh
./build/local/hardsigmoid-test
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21057153](https://our.internmc.facebook.com/intern/diff/D21057153)

[ghstack-poisoned]
